tests/manager/repositories/group/test_admin_repository.py

```diff
@@ -0,0 +1,611 @@
+    ProjectResourcePolicyRow,
+    RoutingRow,
+    ScalingGroupRow,
+    SessionRow,
+    UserResourcePolicyRow,
+    UserRow,
+)
+from ai.backend.manager.models.hasher.types import PasswordInfo
+from ai.backend.manager.models.scaling_group import ScalingGroupOpts
+from ai.backend.manager.models.storage import StorageSessionManager
+from ai.backend.manager.models.user import UserRole, UserStatus
+from ai.backend.manager.models.utils import ExtendedAsyncSAEngine
+from ai.backend.manager.repositories.group.admin_repository import AdminGroupRepository
+
+
+class TestAdminGroupRepositoryDeleteEndpoints:
+    """Test cases for AdminGroupRepository._delete_group_endpoints"""
+
+    @pytest.fixture
+    def test_password_info(self) -> PasswordInfo:
+        """Create a test PasswordInfo object"""
+        return PasswordInfo(
+            password="test_password",
+            algorithm=PasswordHashAlgorithm.PBKDF2_SHA256,
+            rounds=100_000,
+            salt_size=32,
+        )
+
+    @pytest.fixture
+    async def db_with_cleanup(
+        self,
+        database_engine: ExtendedAsyncSAEngine,
+    ) -> AsyncGenerator[ExtendedAsyncSAEngine, None]:
+        """Database engine with cleanup after each test"""
+        yield database_engine
+
+        # Cleanup test data in correct order (respecting FK constraints)
+        async with database_engine.begin_session() as session:
+            # Delete in reverse dependency order: routings → sessions → endpoints → groups → users → policies → domains → scaling_groups
+            await session.execute(
+                sa.delete(RoutingRow),
+                execution_options={"synchronize_session": False},
+            )
+            await session.execute(
+                sa.delete(SessionRow),
+                execution_options={"synchronize_session": False},
+            )
+            await session.execute(
+                sa.delete(EndpointRow),
+                execution_options={"synchronize_session": False},
+            )
+            await session.execute(
+                sa.delete(GroupRow),
+                execution_options={"synchronize_session": False},
+            )
+            await session.execute(
+                sa.delete(UserRow),
+                execution_options={"synchronize_session": False},
+            )
+            await session.execute(
+                sa.delete(ProjectResourcePolicyRow),
+                execution_options={"synchronize_session": False},
+            )
+            await session.execute(
+                sa.delete(UserResourcePolicyRow),
+                execution_options={"synchronize_session": False},
+            )
+            await session.execute(
+                sa.delete(DomainRow),
+                execution_options={"synchronize_session": False},
+            )
+            await session.execute(
+                sa.delete(ScalingGroupRow),
+                execution_options={"synchronize_session": False},
+            )
```

Comment:
There are no conditions in "where", so there is nothing to evaluate, and it doesn't seem necessary to apply "execution_options".